kanidm_unix_int/src/tasks_daemon.rs

```diff
@@ -96,25 +109,37 @@ fn create_home_directory(info: &HomeDirectoryInfo, home_prefix: &str) -> Result<
         return Err("Invalid/Corrupt home directory path - no prefix found".to_string());
     }
 
-    let hd_path_os =
-        CString::new(hd_path_raw.clone()).map_err(|_| "Unable to create c-string".to_string())?;
-
     // Does the home directory exist?
     if !hd_path.exists() {
         // Set a umask
         let before = unsafe { umask(0o0027) };
-        // TODO: Should we copy content from /etc/skel?
         // Create the dir
         if let Err(e) = fs::create_dir_all(hd_path) {
             let _ = unsafe { umask(before) };
             return Err(format!("{:?}", e));
         }
         let _ = unsafe { umask(before) };
 
-        // Change the owner to the gid - remember, kanidm ONLY has gid's, the uid is implied.
-
-        if unsafe { lchown(hd_path_os.as_ptr(), info.gid, info.gid) } != 0 {
-            return Err("Unable to set ownership".to_string());
+        chown(hd_path, info.gid)?;
+
+        // Copy in structure from /etc/skel/ if present
+        let skel_dir = Path::new("/etc/skel/");
+        if skel_dir.exists() {
+            info!("preparing homedir using /etc/skel");
+            for entry in WalkDir::new(skel_dir).into_iter().filter_map(|e| e.ok()) {
+                let dest = &hd_path.join(
+                    entry
+                        .path()
+                        .strip_prefix(skel_dir)
+                        .map_err(|e| e.to_string())?,
+                );
+                if entry.path().is_dir() {
+                    fs::create_dir_all(dest).map_err(|e| e.to_string())?;
+                } else {
+                    fs::copy(entry.path(), dest).map_err(|e| e.to_string())?;
```

Comment:
fs::copy will *overwrite* the content if it already exists, so this will nuke user changes on login.

Reply:
Overwriting shouldn't be an issue if this only fires when the canonical home directory for a user isn't present as noted above

Reply:
Yep, my mistake :)